tasks/pkg-redhat/install-apm-inject.yml

```diff
@@ -0,0 +1,44 @@
+---
+- name: Set APM injection languages and versions for Redhat
+  set_fact:
+    agent_dd_apm_install_pkgs: "{{ (agent_dd_apm_install_pkgs |
+      default([], true)) + ['datadog-apm-library-' + (item | regex_replace('[:]', '-'))] }}"
+  loop: "{{ pre_processed_datadog_apm_libraries }}"
+
+# DNF Handlers -------------------------------------
+- name: Install APM inject library (dnf)
+  dnf:
+    name: 'datadog-apm-inject'
+    update_cache: true
+    state: latest # noqa package-latest
+  when: not ansible_check_mode and ansible_pkg_mgr == "dnf"
+
+- name: Install APM inject libraries (dnf)
+  dnf:
+    name: "{{ (item | default([], true)) }}"
+    state: "{{ 'present' if '-' in item else 'latest' }}" # noqa package-latest
+  loop: "{{ agent_dd_apm_install_pkgs }}" # noqa package-latest
+  when: not ansible_check_mode and ansible_pkg_mgr == "dnf"
+# End DNF Handlers ----------------------------------
+
+# Yum Handlers  -------------------------------------
+- name: Install APM inject library (yum)
+  yum:
+    name: 'datadog-apm-inject'
+    update_cache: true
+    state: latest # noqa package-latest
+  when: not ansible_check_mode and ansible_pkg_mgr == "yum"
+
+- name: Install APM inject libraries (yum)
+  yum:
+    name: "{{ (item | default([], true)) }}"
+    state: "{{ 'present' if '-' in item else 'latest' }}" # noqa package-latest
+  loop: "{{ agent_dd_apm_install_pkgs }}" # noqa package-latest
+  when: not ansible_check_mode and ansible_pkg_mgr == "yum"
+# End Yum Handlers ----------------------------------
+
+- name: Prevent libraries from being upgraded
+  dpkg_selections:
```

Comment:
`dpkg_selections` won't work for RPM-based distros. Please take a look at the install_script example and also at `tasks/pkg-redhat.yml` (look for `includepkgs`) - basically we will have to amend the `includepkgs` at that file with all the libraries that are to be pinned.

Reply:
done